src/utils/getUrlByRequestOptions.ts

```diff
@@ -75,14 +75,7 @@ function getPortByRequestOptions(
   return undefined
 }
 
-function getHostByRequestOptions(options: ResolvedRequestOptions): string {
-  const { hostname, host } = options
-
-  // If the hostname is specified, resolve the host from the "host:port" string.
-  if (hostname != null) {
-    return hostname.replace(/:\d+$/, '')
-  }
-
+function getHostByRequestOptions(host: RequestOptions['host']): string {
```

Comment:
Isn't `host` just `string | undefined`? I think we can annotate that manually, no need to reference the type from `RequestOptions`.

Reply:
I can. Just FYI, this is: `string | null | undefined`

Reply:
I think explicit type is better here. This way if type defs for Node slip in a bug we won't just blindly forward it to the consumer.

```ts
host?: string | null
```

Reply:
Fixed